### PR TITLE
Feature/refactor modules

### PR DESCRIPTION
**Pull request by max-jacobs**

### CODE REVIEW

File: ecs-cleanup-iam.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2018-2019 Expedia, Inc.
```
Comment (patduin) [inline ecs-cleanup-iam.tf]: roaring 20s!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,12 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [1.1.1] - TBD
+## [2.0.0] - TBD
+### Changed
```
Comment (KenFigueiredo) [inline CHANGELOG.md]: Do we need to swap the order of `Changed` and `Added` to adhere to the Keep a Changelog?

File: ecs-cleanup-iam.tf (hunk 1 above)
Comment (max-jacobs) [reply ecs-cleanup-iam.tf]: right you are :), will update

File: variables.tf [hunk 3]
```diff
@@ -13,7 +13,7 @@ variable "instance_name" {
 }
 
 variable "instance_type" {
-  description = "Service to run Beekeeper on. Supported services: `ecs` (default). Leaving this blank will deploy auxilliary components (e.g. RDS, SQS etc.) and will output IAM policies which can used to create roles for your instance type, e.g. EKS."
+  description = "Service to run Beekeeper on. Supported services: `ecs` (default), `k8s`. Leaving this blank will still deploy auxilliary components (e.g. RDS, SQS etc.)."
```
Comment (nvitucci) [inline variables.tf]: Typo on "auxilliary".

File: ecs-templates.tf [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 data "template_file" "beekeeper_path_scheduler_container_definition" {
   count    = var.instance_type == "ecs" ? 1 : 0
-  template = file("${path.module}/files/container-definition.json")
+  template = file("${path.module}/files/ecs-container-definition.json")
```
Comment (KenFigueiredo) [inline ecs-templates.tf]: This is more of a question on the general name changes here - are we only ever going to create `sns` resources for ecs deployments? In those cases what would be the drop in on kubernetes for those?

File: variables.tf [hunk 5]
```diff
@@ -313,107 +313,127 @@ variable "k8s_node_tolerations" {
...
 variable "k8s_ingress_enabled" {
-  description = "Boolean flag to determine if we should create an ingress or not. (0 = off, 1 = on)"
+  description = "Boolean flag to determine if we should create an ingress or not. (0 = off, 1 = on)."
   default     = 0
   type        = number
 }
 
 variable "k8s_ingress_tls_hosts" {
-  description = "List of hosts to apply TLS to the generated kubernetes ingress"
+  description = "List of hosts to apply TLS to the generated kubernetes ingress."
```
Comment (nvitucci) [inline variables.tf]: "Kubernetes" should have a capital K. Maybe it should be rephrased as well?

**Comment by KenFigueiredo** [comment]
Also should we roll https://github.com/ExpediaGroup/beekeeper-terraform/pull/17 into this branch first before we go into master to clear off the CI failures?

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,7 +3,12 @@ All notable changes to this project will be documented in this file.
...
+- Refactored to remove `main` and `lambda` modules.
+- Update all terraform to be compliant with Terraform 0.12 and not throw warnings.
+- Run `terraform fmt` on all code to enforce style.
+
 ### Added
 - Add support for kubernetes by specifying `var.instance_type = "k8s"`
```
Comment (nvitucci) [inline CHANGELOG.md]: Capital "K" for Kubernetes?

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,7 +3,12 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [1.1.1] - TBD
+## [2.0.0] - TBD
+### Changed
+- Refactored to remove `main` and `lambda` modules.
+- Update all terraform to be compliant with Terraform 0.12 and not throw warnings.
```
Comment (nvitucci) [inline CHANGELOG.md]: "Updated all Terraform configurations" - better?

File: CHANGELOG.md (hunk 2 above)
Comment (max-jacobs) [reply CHANGELOG.md]: Nope we don't, will switch this back

File: variables.tf (hunk 3 above)
Comment (max-jacobs) [reply variables.tf]: good spot :)

File: ecs-templates.tf (hunk 4 above)
Comment (max-jacobs) [reply ecs-templates.tf]: I changed the name of "container-definition" to make it distinct from any k8s configuration.

With regards to SNS resources, do you mean the `operational-events` SNS topic? This is only needed for ECS as for ECS we set up some Cloudwatch dashboards which monitor the resources, and need a topic to send events to.

**Pull request merged**